Mocking/README.md

```diff
@@ -38,16 +38,104 @@
 
 ## Mockito basics step-by-step
 
-**TBD**
+In this example project [BookBusinessImpl](src\main\java\com\gepardec\tdd\BookBusinessImpl.java) is the class to be tested.
+It performs operations on book-titles that are retrieved through the [BookService-Interface](src\test\java\com\gepardec\tdd\workdir\BookService.java)
+which represents something we do not have control over like an external service.
 
-Relevant classes: 
- * [BookBusinessImpl](src\main\java\com\gepardec\tdd\BookBusinessImpl.java)
- * [BookService](src\test\java\com\gepardec\tdd\workdir\BookService.java)
- * [BookServiceStub](src\test\java\com\gepardec\tdd\workdir\BookServiceStub.java)
- * [BookBusinessImplStubTest](src\test\java\com\gepardec\tdd\workdir\BookBusinessImplStubTest.java)
- * [BookBusinessImplMockitoTest](src\test\java\com\gepardec\tdd\workdir\BookBusinessImplMockitoTest.java)
- * [BookBusinessImplMockInjectionTest](src\test\java\com\gepardec\tdd\workdir\BookBusinessImplMockInjectionTest.java)
+In the [workdir package](src\test\java\com\gepardec\tdd\workdir) you already find all classes that you will need for completion of this exercise.
 
+**Step 1: Stubbing**
+
+Implement the already in [BookBusinessImplStubTest](src\test\java\com\gepardec\tdd\workdir\BookBusinessImplStubTest.java) present test, which should test
+BookBusinessImpl.retrieveAllJavaBookTitles(...) by using [BookServiceStub](src\test\java\com\gepardec\tdd\workdir\BookServiceStub.java) to control the data returned by the external service.
+
+**Step 2: Mockito**
+
+Now we tackle actually mocking the behaviour of our external service by using Mockito. To create the mock-object we use Mockitos mock method.
+Once the object has been created we also have to specify the behaviour of it. To do so, your code should look something like:
+```java
+Service service = mock(Service.class);
+when(service.call()).thenReturn(data);
+```
+Again implement a test(this time in [BookBusinessImplMockitoTest](src\test\java\com\gepardec\tdd\workdir\BookBusinessImplMockitoTest.java)) on BookBusinessImpl.retrieveAllJavaBookTitles(...) this time by mocking the external service.
+
+**Step 2: BDD**
+
+A popular testing practice that is closely related to TDD is Behaviour Driven Development or
+[BDD](https://www.tricentis.com/blog/bdd-behavior-driven-development/). It's key characteristic is a very descriptive way to write tests that can also
+act as a documentation of your code. It uses a Given-When-Then-Structure when writing tests, for example:
+```java
+// given
+user.setRole(Roles.Admin);
+
+// when
+Response respone = service.call(user);
+
+// then
+response.getReturnCode().isOk;
+```
+Try implementing the same test as in Step 2, but now using a BDD-like structure for your test method.
+Mockito also has methods matching the BDD way of testing:
+```java
```

Comment:
Evtl. vor und nach Code Blöcken Leerzeilen machen, wenn nötig. Das Readme wird vermutlich auch oft in "roher" Form gelesen, nachdem das Training ausgecheckt wurde.